cpp/src/arrow/pretty_print.cc

```diff
@@ -476,8 +479,7 @@ Status PrettyPrint(const ChunkedArray& chunked_arr, const PrettyPrintOptions& op
     } else {
       PrettyPrintOptions chunk_options = options;
       chunk_options.indent += options.indent_size;
-      ArrayPrinter printer(chunk_options, sink);
-      RETURN_NOT_OK(printer.Print(*chunked_arr.chunk(i)));
+      RETURN_NOT_OK(PrettyPrint(*chunked_arr.chunk(i), chunk_options, sink));
```

Comment:
This is creating a new `ArrayPrinter` on every iteration. A cleaner fix here would be changing `Status Print(const Array& array) { ... }` right before `RETURN_NOT_OK(VisitArrayInline(array, this));`

Constructing an `ArrayPrinter` is very cheap at the moment, but that might not forever be the case.